TESTS/mbedmicro-rtos-mbed/mutex/main.cpp

```diff
@@ -12,51 +12,62 @@ using namespace utest::v1;
 
 #define TEST_STACK_SIZE 512
 
-#define TEST_ONE_SEC_MS  (1000)
-#define TEST_HALF_SEC_MS (500)
-#define TEST_HALF_SEC_US (500000)
-#define TEST_ONE_MS_US   (1000)
-
-#define THREAD_DELAY     50
-#define SIGNALS_TO_EMIT  100
+#define TEST_LONG_DELAY 20
+#define TEST_DELAY 10
+#define SIGNALS_TO_EMIT 100
 
 Mutex stdio_mutex;
 
 volatile int change_counter = 0;
 volatile bool changing_counter = false;
 volatile bool mutex_defect = false;
 
-bool manipulate_protected_zone(const int thread_delay) {
+bool manipulate_protected_zone(const int thread_delay)
+{
     bool result = true;
 
     osStatus stat = stdio_mutex.lock();
-    TEST_ASSERT_EQUAL(stat, osOK);
+    TEST_ASSERT_EQUAL(osOK, stat);
+
+    core_util_critical_section_enter();
     if (changing_counter == true) {
         result = false;
         mutex_defect = true;
     }
     changing_counter = true;
 
     change_counter++;
+    core_util_critical_section_exit();
```

Comment:
Now, I'm thinking about, `change_counter` Is the variable under test. It is maybe not the best to protect it with a critical section like other flags.